salsa-2022-tests/tests/accumulate-reuse-workaround.rs

```diff
@@ -89,5 +89,6 @@ fn test1() {
         [
             "accumulated(List(Id { value: 1 }))",
             "compute(List(Id { value: 1 }))",
+            "compute(List(Id { value: 2 }))",
```

Comment:
This is a bug.

> It might seem like this is not valid, because we could still encounter an input later that causes our query to re-execute. But actually I think it is valid, because we know that all the inputs seen thus far to the query were green, and therefore even if the query re-executes, it will store the same value (or else the program is not deterministic).

There are three kinds of output  
* tracked structs created
* invocations of `specify`
* accumulators pushed to

I think you are right about first two cases, but for *accumulators*, things might be different.

take the example as 

```rust
#[salsa::accumulator]
struct Integers(u32);

#[salsa::tracked]
fn compute(db: &dyn Db, input: Input) -> u32 {
    Integers::push(db, 0);
    input.value(db)
}

#[test]
fn test1() {
    let mut db = Database::default();

    let input = Input::new(&mut db, 1);
    dbg!(compute::accumulated::<Integers>(&db, input));

    input.set_value(&mut db).to(2);
    dbg!(compute::accumulated::<Integers>(&db, input));
}
```
will output

```
[salsa-2022-tests/tests/accumulate-reuse-workaround.rs:53] compute::accumulated::<Integers>(&db, input) = [
    0,
]
[salsa-2022-tests/tests/accumulate-reuse-workaround.rs:56] compute::accumulated::<Integers>(&db, input) = [
    0,
    0,
]
```
If it was the first push in a new version, we would reset the accumulated values by checking the `produced_at` of the old accumulated values

https://github.com/salsa-rs/salsa/blob/07848c2818077635658ba94b245d6f5713909452/components/salsa-2022/src/accumulator.rs#L64-L68

But now we will first mark the old accumulated values as valid in `deep_verify_memo`, which will update the `produced_at' of the old accumulated values to the newest version so that when we re-execute the query and push the values, we will not reset them.

Reply:
I see, yes. Hmm. The problem here is that it's not a single, instantaneous write, but rather a series of writes. Let me ponder this for a second.